### PR TITLE
feat(absolute-time-picker): remove date input limits inplace validation

### PR DESCRIPTION
**Pull request by djcouto**
Screenshot 2024-04-05 at 16 01 58
Screenshot 2024-04-05 at 16 01 43
Screenshot 2024-04-05 at 16 01 27

### CODE REVIEW

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts [hunk 1]
```diff
@@ -45,3 +45,67 @@ export const getMaximumDateFromDayClick = (clickedDate: dayjs.Dayjs, maximumDate
...
+	{ minDate, maxDate }: IAbsoluteTimeLimits
+): { state: EValidationState; helpText: string } | undefined => {
+	if (!error) {
+		return;
+	}
+
+	if (error === EAbsoluteTimeError.StartDateBeforeMinimumDate) {
+		return {
+			state: EValidationState.Invalid,
+			helpText: `The date must be after ${minDate}`
```
Comment (GoncaloCamaz) [inline packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts:99]: Same here and bellow?

**Comment by GoncaloCamaz** [comment]
Great improvement! 🚀

**Review by ricardogoncalves89 — APPROVED** [review]

File: packages/ui-components/src/components/absolute-time-picker-dropdown/absolute-time-picker-dropdown.utils.ts [hunk 2]
```diff
@@ -36,7 +36,43 @@ export const getFormattedSelectedDates = (range: SelectedTimestamp, mode: EAbsol
 	return [dateFrom, dateTo];
 };
 
-export const areDatesValidByRange = (range: SelectedTimestamp, mode: EAbsoluteTimePickerMode): boolean => {
+export const getAbsoluteTimePickerError = (range: SelectedTimestamp, mode: EAbsoluteTimePickerMode, limits: IAbsoluteTimeLimits): EAbsoluteTimeError => {
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/absolute-time-picker-dropdown/absolute-time-picker-dropdown.utils.ts:39]: The function signature also should mark `undefined` as a possible return value

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts [hunk 3]
```diff
@@ -45,3 +46,69 @@ export const getMaximumDateFromDayClick = (clickedDate: dayjs.Dayjs, maximumDate
 
 	return clickedDate.endOf('day');
 };
+
+export const getFromDateInputState = (error: EAbsoluteTimeError | undefined, { minDate }: IAbsoluteTimeLimits): { state: EValidationState; helpText: string } | undefined => {
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts:50]: should do we define a type for this `{ state: EValidationState; helpText: string } | undefined`?

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts [hunk 4]
```diff
@@ -45,3 +46,69 @@ export const getMaximumDateFromDayClick = (clickedDate: dayjs.Dayjs, maximumDate
...
+	if (!error) {
+		return;
+	}
+
+	if (error === EAbsoluteTimeError.StartDateBeforeMinimumDate) {
+		const min = (isNumber(minDate) ? dayjs(minDate) : dayjs(minDate, DATETIME_INPUT_MASK)).format(DATETIME_INPUT_MASK);
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts:56]: can we normalize the min e max value to be always a number?

**Review by miguelpinto98 — COMMENTED** [review]
All good, just left some questions to discuss before the approval.

**Review by GoncaloCamaz — APPROVED** [review]

File: packages/ui-components/src/components/absolute-time-picker-dropdown/absolute-time-picker-dropdown.utils.ts (hunk 2 above)
Comment (djcouto) [reply packages/ui-components/src/components/absolute-time-picker-dropdown/absolute-time-picker-dropdown.utils.ts:39]: Done 👍

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts (hunk 3 above)
Comment (djcouto) [reply packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts:50]: Done 👍

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts (hunk 4 above)
Comment (djcouto) [reply packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.helper.ts:56]: Done 👍

**Review by miguelpinto98 — APPROVED** [review]
👍

**Pull request merged**